numba/core/ir_utils.py

```diff
@@ -2256,3 +2266,43 @@ def convert_code_obj_to_function(code_obj, caller_ir):
     # create the function and return it
     return _create_function_from_code_obj(fcode, func_env, func_arg, func_clo,
                                           glbls)
+
+
+def fixup_var_define_in_scope(blocks):
+    """Fixes the mapping of ir.Block to ensure all ir.Var are defined in all
+    scope.
```

Comment:
```suggestion
    """Fixes the mapping of ir.Block to ensure all ir.Var within a given block
    are defined in the block's associated scope.
```

Reply:
It's actually making sure that all the scopes referenced in the function know about all the variables used in the function.

Reply:
I tried to clarify it in the latest commit.

Reply:
Thanks, latest comment looks good.